### PR TITLE
[wol] Implement wol command line utility

### PR DESCRIPTION
**Pull request by lizhijianrd**
#### What I did

Implement `wol` command-line utility based on [Wake-on-LAN-HLD.md](https://github.com/sonic-net/SONiC/blob/master/doc/wol/Wake-on-LAN-HLD.md).

#### How to verify it

Add unittest to verify code.

### CODE REVIEW

File: wol/main.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            password = socket.inet_aton(value)
+        except OSError:
+            raise click.BadParameter("invalid password format")
+    else:  # The length of a valid MAC address is 17.
+        try:
+            password = binascii.unhexlify(value.replace(':', '').replace('-', ''))
+        except binascii.Error:
+            raise click.BadParameter("invalid password format")
+    if len(password) not in [4, 6]:
+        raise click.BadParameter("password must be 4 or 6 bytes")
```
Comment (Blueve) [inline wol/main.py:140]: this statement is conflicting with line 128

File: wol/main.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    return password
```
Comment (Blueve) [inline wol/main.py:141]: The function is unlikely just `validate` the password, it parsed the password.
Suggest change name to `parse_password`

File: wol/main.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+def validate_interface(ctx, param, value):
+    if value not in netifaces.interfaces():
+        raise click.BadParameter("invalid SONiC interface name {}".format(value))
+    if get_interface_operstate(value) != 'up':
+        raise click.BadParameter("interface {} is not up".format(value))
+    return value
+
+
+def validate_target_mac(ctx, param, value):
```
Comment (Blueve) [inline wol/main.py:113]: `parse_target_mac`

File: wol/main.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    mac_list = []
+    for mac_s in value.split(','):
+        try:
+            mac_b = binascii.unhexlify(mac_s.replace(':', '').replace('-', ''))
```
Comment (Blueve) [inline wol/main.py:117]: what is the `_b` means?

File: wol/main.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+
+def get_interface_operstate(interface):
+    with open('/sys/class/net/{}/operstate'.format(interface), 'r') as f:
+        return f.read().strip().lower()
+
+
+def get_interface_mac(interface):
+    mac = netifaces.ifaddresses(interface)[netifaces.AF_LINK][0].get('addr')
+    return MacAddress(binascii.unhexlify(mac.replace(':', '').replace('-', '')))
```
Comment (Blueve) [inline wol/main.py:80]: There are many places you use this pattern:
```
binascii.unhexlify(mac.replace(':', '').replace('-', '')
```

can you extract a method for it?

**Review by Blueve — CHANGES_REQUESTED** [review]

File: wol/main.py (hunk 1 above)
Comment (lizhijianrd) [reply wol/main.py:140]: line 128 `return` directly, will not hit this line.

File: wol/main.py (hunk 2 above)
Comment (lizhijianrd) [reply wol/main.py:161]: Good catch! updated the function name.

File: wol/main.py (hunk 3 above)
Comment (lizhijianrd) [reply wol/main.py]: Updated.

File: wol/main.py (hunk 4 above)
Comment (lizhijianrd) [reply wol/main.py]: I used this suffix to describe the variable type is bytes. I've refactored the code of class `MacAddress`. No longer need this suffix now.

File: wol/main.py (hunk 5 above)
Comment (lizhijianrd) [reply wol/main.py]: Thanks for pointing out. I've refactor the `MacAddress` class to integrate this logic.

File: wol/main.py (hunk 1 above)
Comment (Blueve) [reply wol/main.py:158]: I mean the wording. You said that *must be* 4 or 6, but 0 is also acceptable based on your implementation.

Comment (lizhijianrd) [reply wol/main.py]: I see. Updated.

**Review by Blueve — APPROVED** [review]

**Comment by lizhijianrd** [comment]
Link sonic-net/SONiC#1508

**Pull request merged**